lib/matplotlib/axes/_axes.py

```diff
@@ -2292,6 +2292,14 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
         xerr = kwargs.pop('xerr', None)
         yerr = kwargs.pop('yerr', None)
         error_kw = kwargs.pop('error_kw', {})
+        ezorder = error_kw.pop('zorder', None)
```

Comment:
We don't need this `pop` if we are going to use `setdefault`.

I think the question is do we want to `.bar(..., error_kw={'zorder': None})` to mean "please do not adjust the zorder and fall through the current behavior" or do we want it to me "do the new default thing and adjust the zorder".

I think we want the second here, but see arguments either way.

Reply:
:+1: for the second.

Reply:
I think for the second option (for which I am also 👍 ) the `pop` needs to stay?